tests/Unit/NumericalAlgorithms/LinearOperators/Test_PartialDerivatives.cpp

```diff
@@ -120,6 +120,34 @@ using two_vars = tmpl::list<Var1<Dim>, Var2>;
 template <size_t Dim>
 using one_var = tmpl::list<Var1<Dim>>;
 
+template <typename GradientTags, typename VariableTags, typename TagsList,
+          size_t Dim>
+void test_logical_partial_derivative(
+    const std::array<Variables<TagsList>, Dim>& du,
+    const Variables<VariableTags>& u, const Mesh<Dim>& mesh) {
+  tmpl::for_each<GradientTags>([&du, &mesh, &u](auto gradient_tag_v) noexcept {
+    using gradient_tag = tmpl::type_from<decltype(gradient_tag_v)>;
+    const auto single_du =
+        logical_partial_derivative(get<gradient_tag>(u), mesh);
+    for (size_t storage_index = 0; storage_index < get<gradient_tag>(u).size();
+         ++storage_index) {
+      for (size_t d = 0; d < Dim; ++d) {
+        const auto deriv_tensor_index =
+            prepend(get<gradient_tag>(u).get_tensor_index(storage_index), d);
+        CHECK_ITERABLE_APPROX(single_du.get(deriv_tensor_index),
+                              get<gradient_tag>(gsl::at(du, d))[storage_index]);
+      }
+    }
+    std::decay_t<decltype(single_du)> single_du_not_null{};
+    DataVector buffer{mesh.number_of_grid_points()};
+    gsl::span<double> buffer_view{buffer.data(), buffer.size()};
+    logical_partial_derivative(make_not_null(&single_du_not_null),
```

Comment:
This only tests the case when the tensor is actually in a contiguous memory block allocated by a `Variables` I think. It should probably also test the case where the tensor is stand-alone, with components potentially in different memory blocks (when the `DataVectors` own the memory)